ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fullsync/FullSyncTargetManager.java

```diff
@@ -105,6 +109,9 @@ private boolean isSyncTargetReached(final EthPeer peer) {
 
   @Override
   public boolean shouldContinueDownloading() {
-    return true;
+    return terminalTotalDifficulty.isEmpty()
+        || terminalTotalDifficulty
+            .get()
+            .greaterThan(protocolContext.getBlockchain().getChainHead().getTotalDifficulty());
```

Comment:
I can't help but want to suggest:

```
    return terminalTotalDifficulty.filter(ttd -> 
        ttd.lessThan(
            protocolContext
                .getBlockchain()
                .getChainHead()
                .getTotalDifficulty()))
        .isPresent();

```

But perhaps yours is more readable

Reply:
I am migrating this part to be more flexible. The terminating condition will be possible to change to a target block hash and even change it during the sync.